hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -2303,6 +2303,7 @@ private <T extends WithParentObjectId> DeleteKeysResult gatherSubPathsWithIterat
           consumedSize += objectSerializedSize;
         }
       }
+      keyInfos.replaceAll(keyInfo -> new OmKeyInfo.Builder(keyInfo).setAcls(Collections.emptyList()).build());
```

Comment:
This is still iterating `keyInfos` again.  To achieve what @aswinshakil suggested, make the replacement at:

https://github.com/apache/ozone/blob/dc3a4aaee6610ec94d307d454dd33a4b11014c93/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java#L2301

Reply:
My bad, done!